lib/services/salesforce.rb

```diff
@@ -11,20 +11,30 @@ class AhaServices::Salesforce < AhaService
   
   install_button
 
+  def receive_updated
+    # Only do the validation if there is something in the host field. This
+    # prevents errors from showing up when their first create the integration.
+    validate_host if data["host"]
+  end
+
   def receive_installed
-    # Validate authentication.
-    client.user_info
-    
-    # Update settings.
-    client.post('/services/apexrest/ahaapp/aha_rest_api/settings', 
-      idea_portal_url: data.idea_portal_url, 
-      jwt_secret_key: data.jwt_secret_key)
-  rescue Exception => e
-    logger.debug("Salesforce authentication problem #{e.class}: #{e.message} #{e.backtrace.join("\n")}")
-    if e.message.include? 'The REST API is not enabled for this Organization.'
-      raise ConfigurationError, "The REST API is not enabled for this Organization."
-    else
-      raise ConfigurationError, "Authentication failed. Please use the 'Authenticate' button to connect to Salesforce."
+    validate_host or raise ConfigurationError, "Please ensure the custom host ends in salesforce.com and do not include http:// or https://."
```

Comment:
Tiny complaint, can we make this `does not include` instead of `do not include`? (Flows better grammatically)